runtime/src/bank_forks.rs

```diff
@@ -184,7 +184,17 @@ impl BankForks {
             last_accounts_hash_slot: root,
             in_vote_only_mode: Arc::new(AtomicBool::new(false)),
             highest_slot_at_startup: 0,
-        }
+        };
+        let root_bank = bank_forks
+            .banks
+            .get(&root)
+            .expect("root bank didn't exist in bank_forks");
+        root_bank
+            .loaded_programs_cache
+            .write()
+            .unwrap()
+            .prune(&bank_forks, root);
```

Comment:
won't this only set the root for the root bank? It seems possible we have more than one bank, which do not (to my knowledge) share `LoadedPrograms`

Reply:
Is that for tests, or do we have this scenario even for a validator?

Reply:
If all non-root banks are constructed via `_from_parent` this is true, so I think you're right for validator/ledger-tool.

I'm not sure about for our tests, where I think we sometimes initialize with multiple banks, not sure if they are directly constructed or descended from common parent. If directly, then they would have independent `LoadedPrograms`